src/e2e-test/java/io/cdap/plugin/utils/PubSubClient.java

```diff
@@ -36,6 +56,22 @@ public static Topic createTopic(String topicId) throws IOException {
     }
   }
 
+  // Create the subscription
+  public static void createSubscription(String subscriptionId, String topicId) throws IOException {
+    ProjectSubscriptionName subscriptionName = null;
+    try (SubscriptionAdminClient subscriptionAdminClient = SubscriptionAdminClient.create(
+      SubscriptionAdminSettings.newBuilder().build())) {
+      TopicName topicName = TopicName.of(PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID), topicId);
+       subscriptionName = ProjectSubscriptionName.of
+         (PluginPropertyUtils.pluginProp(ConstantsUtil.PROJECT_ID), subscriptionId);
+      subscriptionAdminClient.createSubscription(subscriptionName, topicName, PushConfig.getDefaultInstance(), 60);
+      System.out.println("Subscription created: " + subscriptionName.toString());
+    } catch (AlreadyExistsException e) {
```

Comment:
Can you try with **StatusRuntimeException**

Reply:
it works with this only for messaging